LIPS/LIP-draft_on_coverage_app.md

```diff
@@ -0,0 +1,465 @@
+---
+lip: TBA
+title: In-protocol coverage
+status: Draft
+author: Sam Kozin, Eugene Mamin
+discussions-to: TBA
+created: TBA
+updated: TBA
+---
+
+# In-protocol coverage proposal
+
+## Summary
+
+Specifies an in-protocol mechanism of applying coverage to address slashing/penalties events.
+
+## Motivation
+
+Currently, Lido has no adopted and well-defined mechanism of applying coverage for stakeholders' losses due to validators pentalties, slashing and other conditions. We propose an in-protocol solution to precisely specify coverage application without breaking existing principles, agreements, and integrations with stETH token. The proposal is aimed to improve the overall technical transparency and completeness of the Lido protocol regarding applying coverage.
+
+## Mechanics
+
+We propose to provide an on-chain in-protocol mechanism of applying coverage by burning stETH token. It relies on the rebasing nature of the stETH. The basic account [balance calculation](https://docs.lido.fi/contracts/lido#rebasing) defined by stETH contract is the following:
+```solidity
+balanceOf(account) = shares[account] * totalPooledEther / totalShares
+```
+So burning someone's shares (e.g. decreasing `totalShares` count) leads to increasing all of the other accounts' balances. 
+
+We propose to deploy a dedicated `LidoStETHBurner` contract which accepts burning requests by locking caller-provided stETH tokens. Those burning requests are initially set by the contract to a pending state. Actual burning happens within an oracle (`LidoOracle`) beacon report to prevent additional fluctuations of the existing stETH token rebase period.
```

Comment:
Maybe rename to something more neutral like `CoverageApplicant`?

Reply:
Yeeep, It's arguable naming questions :smile: 

Our proposal also supports non-coverage application cases, so explicitly using "Coverage*" may confuse and mislead someone, I think.

To suggest another one option, maybe ```LidoStEthValueRecoverer``` sounds good for you?

Reply:
I'd leave it as is since all this contract does is burning stETH, and it is not concerned about anything else.

Reply:
Also, as per the proposal, we want to make this contract the only address allowed to burn stETH.

Reply:
I see how having smth that's called "stETH burner" in the protocol might frighten someone but I'd opt for transparency here. We can add a large comment on to the top of that contract explaining that it's not allowed to burn anyone else's stETH and is used for distributing value among stETH holders which is most easily done by burning some stETH.

Reply:
This mechanism of value distribution between token holders is well known among the Ethereum community since it's used not only by Ethereum itself (EIP-1559) but by other large protocols and even CEXes.